src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -59,7 +62,7 @@ class Config:
             {'prefix': self._config.datastore_prefix}
         )
 
-        await self.init_shutdown_script_extend(new)
+        await self.middleware.call('system.update_timeout_value')
```

Comment:
You missed delete.
But how about do the other way around and use the initshutdownscript.post_{create,update,delete} hooks instead in system plugin and keep it all in one place?

Reply:
This is really nice :) I wasn't aware of those before. I was thinking we should have hooks of some sort. I'll have this covered right away. Thank you